src/plugins/intel_gna/src/gna_graph_compiler.cpp

```diff
@@ -663,6 +663,99 @@ void GNAGraphCompiler::finalizeConvolution1DPrimitive(InferenceEngine::CNNLayerP
+            return create_transpose_weights(extended_weights.data());
+        }
+
+        return create_transpose_weights(m_data.orginal_kernel_data);
+    }
+
+private:
+    bool is_extendind_weights_needed() {
+        if (m_data.original_kernel_height != m_data.effective_kernel_height ||
+            m_data.effective_kernel_width != m_data.effective_kernel_height) {
+            return true;
+        }
+        return false;
+    }
+
+    std::vector<uint8_t> create_extended_weights() {
+        const auto original_kernel_hw = m_data.original_kernel_width * m_data.original_kernel_height;
+        const auto original_kernel_size = m_data.input_channels * original_kernel_hw * m_data.convolution_precision;
+        const auto original_kernel_line =
+            m_data.input_channels * m_data.original_kernel_width * m_data.convolution_precision;
+        const auto orginal_kernel_channel_size = m_data.original_kernel_width * m_data.convolution_precision;
+
+        const auto new_kernel_channel_size = m_data.effective_kernel_width * m_data.convolution_precision;
+        const auto new_kernel_line_size = m_data.input_channels * new_kernel_channel_size;
+
+        const auto new_kernel_hw = m_data.effective_kernel_width * m_data.effective_kernel_height;
+        const auto new_kernel_size = m_data.input_channels * new_kernel_hw * m_data.convolution_precision;
+
+        const auto kernels_num = m_data.kernels_num;
+        const auto weights_size = kernels_num * new_kernel_size;
+
+        std::vector<uint8_t> weights(weights_size, 0);
+
+        for (uint32_t k = 0; k < kernels_num; k++) {
+            uint8_t* current_source_filter = m_data.orginal_kernel_data + k * original_kernel_size;
+            for (uint32_t j = 0; j < m_data.effective_kernel_height; ++j) {
+                uint8_t* current_source_filter_row = current_source_filter + j * original_kernel_line;
+                if (j < m_data.original_kernel_height) {
+                    for (int i = 0; i < m_data.input_channels; ++i) {
+                        auto ptr_line_start = current_source_filter_row + i * orginal_kernel_channel_size;
+                        auto ptr_line_end = ptr_line_start + orginal_kernel_channel_size;
+                        std::copy(ptr_line_start,
+                                  ptr_line_end,
+                                  weights.begin() + k * new_kernel_size + j * new_kernel_line_size +
+                                      i * new_kernel_channel_size);
+                    }
+                }
+            }
+        }
+        return weights;
+    }
+
+    std::vector<uint8_t> create_transpose_weights(uint8_t* src_weights) {
+        const auto kernel_hw = m_data.effective_kernel_width * m_data.effective_kernel_height;
+        const uint32_t kernel_size = m_data.input_channels * kernel_hw * m_data.convolution_precision;
+        const auto kernel_pad = Gna2RoundUp(kernel_size, 16) - kernel_size;
```

Comment:
Everywhere else, plugin is using ALIGN instead of Gna2RoundUp
```suggestion
        const auto kernel_pad = ALIGN(kernel_size, 16) - kernel_size;
```